pipelinit/pipelinit-cli
# feat(core/cli) add docker release stage with sensible default

Following the discussion, https://github.com/pipelinit/pipelinit-cli/issues/34;
this PR implements a release stack for docker images that can be built in the repository and
with that adds a sensitive default to the Pipelinit configuration file with the possibility to other 'registries' for the generated CI.

The template basically looks for paths with Dockerfiles that can be built and
adds metadata involving the path found and the specified version.

Resolves: https://github.com/pipelinit/pipelinit-cli/issues/80

To test:
- Run Pipelinit on a project with a docker file, then configure the secrets accordingly on GitHub and check if the generated pipeline correctly push the image to the registries.

Example:
- Using `ghcr.io`:
  - https://github.com/orgs/pipelinit/packages?repo_name=pipelinit-sample-docker
  - https://github.com/pipelinit/pipelinit-sample-docker/runs/3899104124?check_suite_focus=true
- Using `registry.hub.docker.com`:
  - https://hub.docker.com/u/jjvlima
  - https://github.com/pipelinit/pipelinit-sample-docker/runs/3898871153?check_suite_focus=true

![image](https://user-images.githubusercontent.com/30262244/137389940-7fb74965-9152-487d-b096-8be2510ee6ac.png)
![image](https://user-images.githubusercontent.com/30262244/137389986-9ec62ad5-793e-4f0e-9b40-ddca993cd72b.png)
